src/getChildEventSubscriber.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @flow
+ */
+
+import type {
+  NavigationEventSubscriber,
+  NavigationAction,
+  NavigationState,
+  NavigationEventPayload,
+} from './TypeDefinition';
+
+/*
+ * This is used to extract one children's worth of events from a stream of navigation action events
+ *
+ * Based on the 'action' events that get fired for this navigation state, this utility will fire
+ * focus and blur events for this child
+ */
+export default function getChildEventSubscriber(
+  addListener: NavigationEventSubscriber,
+  key: string
+): NavigationEventSubscriber {
+  const actionSubscribers = new Set();
+  const willFocusSubscribers = new Set();
+  const didFocusSubscribers = new Set();
+  const willBlurSubscribers = new Set();
+  const didBlurSubscribers = new Set();
+
+  const getChildSubscribers = evtName => {
+    switch (evtName) {
+      case 'action':
```

Comment:
note: in follow-up pr, we should make sure that an event fires for mayBlur here (or whatever we want to call it) so we can hide keyboard etc when we start dragging